cluster-autoscaler/cloudprovider/cloudifyprovider/scale_provider.go

```diff
@@ -17,37 +17,97 @@ limitations under the License.
 package cloudifyprovider
 
 import (
-	cloudify "github.com/cloudify-incubator/cloudify-rest-go-client/cloudify"
+	"github.com/cloudify-incubator/cloudify-rest-go-client/cloudify"
 	"github.com/golang/glog"
 	apiv1 "k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/api/resource"
 	"k8s.io/autoscaler/cluster-autoscaler/cloudprovider"
 	"k8s.io/autoscaler/cluster-autoscaler/utils/errors"
+	"fmt"
 )
 
 // CloudifyScaleProvider - settings for connect to cloudify
 type CloudifyScaleProvider struct {
 	client          *cloudify.Client
-	deploymentID    string
+	deployment      string
 	resourceLimiter *cloudprovider.ResourceLimiter
 }
 
+func (clsp *CloudifyScaleProvider) GetDeploymentNodeInfo() ([]map[string]string, error) {
+	data, err := cloudify.ParseDeploymentFile(clsp.deployment)
+	deploymentList := []map[string]string{}
+	if err != nil {
+		fmt.Errorf("Error While trying to parse deployment file")
+		return nil, err
+	}
+
+	for _, deployment := range data.Deployments {
+		deploymentInfo := make(map[string]string)
+		dep := deployment.(map[string]interface{})
+
+		deploymentInfo["id"] = dep["id"].(string)
+		deploymentInfo["node_data_type"] = dep["node_data_type"].(string)
+		deploymentInfo["deployment_type"] = dep["deployment_type"].(string)
+		deploymentList = append(deploymentList, deploymentInfo)
+	}
+
+	return deploymentList, nil
+}
+
 // Name returns name of the cloud provider.
 func (clsp *CloudifyScaleProvider) Name() string {
-	glog.V(4).Infof("Name: %v", clsp.deploymentID)
+	items, err := clsp.GetDeploymentNodeInfo()
```

Comment:
Looks as debug code, could we drop it?

Reply:
We don't need it since I change how we handle deployment and no need to have it here
@0lvin

Reply:
@mabuaisha So remove :-)